sklearn/utils/_tags.py

```diff
@@ -1,71 +1,318 @@
+        to change because we are trying to make it more flexible: be
+        prepared for breaking changes in the future.
+
+    input_tags : InputTags
+        The input data(X) tags.
+    """
+
+    target_tags: TargetTags
+    transformer_tags: TransformerTags | None
+    classifier_tags: ClassifierTags | None
+    regressor_tags: RegressorTags | None
+    array_api_support: bool = False
+    no_validation: bool = False
+    non_deterministic: bool = False
+    requires_fit: bool = True
+    _skip_test: bool = False
+    _xfail_checks: dict[str, str] = field(default_factory=dict)
+    input_tags: InputTags = field(default_factory=InputTags)
+
+
+def default_tags(estimator) -> Tags:
+    """Get the default tags for an estimator.
+
+    This ignores any ``__sklearn_tags__`` method that the estimator may have.
+
+    If the estimator is a classifier or a regressor, ``target_tags.required``
+    will be set to ``True``, otherwise it will be set to ``False``.
+
+    ``transformer_tags`` will be set to ``TransformerTags()`` if the estimator
+    has a ``transform`` or ``fit_transform`` method, otherwise it will be set
+    to ``None``.
+
+    ``classifier_tags`` will be set to ``ClassifierTags()`` if the estimator is
+    a classifier, otherwise it will be set to ``None``.
+
+    ``regressor_tags`` will be set to ``RegressorTags()`` if the estimator is a
+    regressor, otherwise it will be set to ``None``.
+
+    Parameters
+    ----------
+    estimator : estimator object
+        The estimator for which to get the default tags.
+
+    Returns
+    -------
+    tags : Tags
+        The default tags for the estimator.
+    """
+    from ..base import is_classifier, is_regressor
+
+    target_required = is_classifier(estimator) or is_regressor(estimator)
+
+    return Tags(
+        target_tags=TargetTags(required=target_required),
+        transformer_tags=(
+            TransformerTags()
+            if hasattr(estimator, "transform") or hasattr(estimator, "fit_transform")
+            else None
+        ),
+        classifier_tags=ClassifierTags() if is_classifier(estimator) else None,
+        regressor_tags=RegressorTags() if is_regressor(estimator) else None,
+    )
+
+
+def get_tags(estimator) -> Tags:
+    """Get estimator tags.
 
     :class:`~sklearn.BaseEstimator` provides the estimator tags machinery.
     However, if an estimator does not inherit from this base class, we should
     fall-back to the default tags.
 
     For scikit-learn built-in estimators, we should still rely on
-    `self._get_tags()`. `_safe_tags(est)` should be used when we are not sure
-    where `est` comes from: typically `_safe_tags(self.base_estimator)` where
-    `self` is a meta-estimator, or in the common checks.
+    `self.__sklearn_tags__()`. `get_tags(est)` should be used when we
+    are not sure where `est` comes from: typically
+    `get_tags(self.estimator)` where `self` is a meta-estimator, or in
+    the common checks.
```

Comment:
When reading this, I'm unsure what the alternative that one would mistakenly do instead of `get_tags(est)`? Is there a naive approach that would result in an error?

Reply:
The alternative is to call `est.__sklearn_tags__` assuming the instance has the method. The `get_tags` method handles cases where that method is not present.